### DISCUSSION TITLE
Keep up the good work!

### DISCUSSION THREAD

**Discussion by RylanYancey** [KyleMayes/vulkanalia#218] [General]
Just wanted to say that you're doing an AWESOME job! I love the builders you've implemented for the config structs, it allows my code to be so much more elegant.  I like how almost everything in the crate is 1:1 with regular vulkan, so I can use existing API documentation to understand the concepts. I'm using Vulkanalia to build a voxel game engine.  

Two questions: 

1. Will Vulkanalia ever have support for VMA? The vk-mem crate expects Ash types for everything, and doesnt compile. The gpu-allocator crate also expects ash. 

2. Can we expect an Egui integration at some point? I'm trying to build my own but it is very poorly documented.

**Comment by KyleMayes** [comment]
Thanks!

I looked a little into what it would take to add support for these two integrations.

An `egui` integration looks like it wouldn't be too big of a lift, having taken a quick look at some existing integrations.
I don't have any specific plans here (and am definitely not making any promises!) but I will try to look into it more soon.

VMA looks a bit more complicated.
I'll need to take a closer look at what `vk-mem` is doing, but I imagine there is a way, with some code generation shenanigans using `bindgen`, to auto-generate an interface to VMA that uses `vulkanalia` types.
I'll need to look into it more.

**Comment by lebedec** [comment]
Hi there! 

I want to join the topic starter. Thanks for your work!

I have already tried to use Vulkan via `ash` and `vulkano` with the official tutorials. But each time it caused a lot of frustration. Your version of the tutorial is the best. Thanks to you, it finally became clear how to work with Vulkan API step by step. I'm using Vulkanalia to create my new 2d game.

Keep up the good work 🚀

**Comment by KyleMayes** [comment]
I've started work to add a VMA integration and wrapper.
The work is being tracked in https://github.com/KyleMayes/vulkanalia/issues/276.
I'll look into `egui` after that work is done.
